### PR TITLE
Implement PgGate_InitPrimaryCluster Api and add config to talk to k2

### PR DESCRIPTION
**Pull request by ahsank**
Currently it creates collection in k2 but boost future in skv client causes sigfault when processing response.

Current Implementation of `PgGate_InitPrimaryCluster` is just to test connectivity. The call to skv client may be moved to a different layer later.

### CODE REVIEW

File: simpleInstall/pg_run.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# Run following in another terminal
+# cd chogori-platform
+#  DARGS='-p 30000:30000' k2build
+# cd test/integration
+# in test_http_client.sh remove last line that runs test and put a wait
+# Run ./test_http_client
+# After that k2 system can be accessed by port 30000
+
+export SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+# export LD_LIBRARY_PATH=$SCRIPT_DIR/../build/src/k2/connector/common/:$SCRIPT_DIR/../build/src/k2/connector/entities/:$SCRIPT_DIR/../src/k2/postgres/lib
```
Comment (johnfangAFW) [inline simpleInstall/pg_run.sh:4]: please remove commented code

File: src/gausskernel/storage/access/k2/pg_gate_api.cpp [hunk 2]
```diff
@@ -46,9 +47,17 @@ std::atomic<bool> api_impl_shutdown_done;
 
 } // anonymous namespace
 
+
+k2pg::catalog::SqlCatalogClient* GetCatalog() {
+    static auto catalogManager = std::make_shared<k2pg::catalog::SqlCatalogManager>();
+    static auto catalog = new k2pg::catalog::SqlCatalogClient(catalogManager);
```
Comment (johnfangAFW) [inline src/gausskernel/storage/access/k2/pg_gate_api.cpp]: why not use smart pointer here?

File: src/gausskernel/storage/access/k2/pg_gate_api.cpp [hunk 3]
```diff
@@ -119,7 +128,25 @@ K2PgStatus PgGate_GetSharedCatalogVersion(uint64_t* catalog_version) {
 K2PgStatus PgGate_InitPrimaryCluster()
 {
   elog(DEBUG5, "PgGateAPI: PgGate_InitPrimaryCluster");
-  return K2PgStatus::NotSupported;
+
+  auto catalog = GetCatalog();
+  auto skvstat = catalog->InitPrimaryCluster();
+  if (skvstat.is2xxOK()) {
+      return K2PgStatus {
```
Comment (johnfangAFW) [inline src/gausskernel/storage/access/k2/pg_gate_api.cpp]: use Status::OK or extend it to pass in the code

File: src/gausskernel/storage/access/k2/pg_gate_api.cpp [hunk 4]
```diff
@@ -119,7 +128,25 @@ K2PgStatus PgGate_GetSharedCatalogVersion(uint64_t* catalog_version) {
...
+          .pg_code = ERRCODE_SUCCESSFUL_COMPLETION,
+          .k2_code = skvstat.code,
+          .msg = "ok",
+          .detail = "ok"
+      };
+  }
+  K2PgStatus status {
+      .pg_code = ERRCODE_FDW_OPERATION_NOT_SUPPORTED,
```
Comment (johnfangAFW) [inline src/gausskernel/storage/access/k2/pg_gate_api.cpp]: suggest to use Status::NotSupported

File: simpleInstall/pg_run.sh (hunk 1 above)
Comment (ahsank) [reply simpleInstall/pg_run.sh:4]: Fixed.

File: src/gausskernel/storage/access/k2/pg_gate_api.cpp (hunk 3 above)
Comment (ahsank) [reply src/gausskernel/storage/access/k2/pg_gate_api.cpp]: Done.

File: src/gausskernel/storage/access/k2/pg_gate_api.cpp (hunk 4 above)
Comment (ahsank) [reply src/gausskernel/storage/access/k2/pg_gate_api.cpp]: Done.

File: src/gausskernel/storage/access/k2/pg_gate_api.cpp (hunk 2 above)
Comment (ahsank) [reply src/gausskernel/storage/access/k2/pg_gate_api.cpp]: Changed to use shared pointer.

**Review by johnfangAFW — APPROVED** [review]

**Pull request merged**